### PR TITLE
add IGNORE rule into analyzer (issue #570)

### PR DESCRIPTION
**Pull request by nikita-moor**
Issue #570. Nearly literal copy of "ignoredchars" code from `HunspellImpl::checkword` function.

### CODE REVIEW

File: src/hunspell/hunspell.cxx [hunk 1]
```diff
@@ -235,7 +235,20 @@ size_t HunspellImpl::cleanword2(std::string& dest,
   dest.clear();
   dest_utf.clear();
 
-  const char* q = src.c_str();
+  std::string w2;
+  w2.assign(src);
+  const char* ignoredchars = pAMgr ? pAMgr->get_ignore() : NULL;
```
Comment (phajdan) [inline src/hunspell/hunspell.cxx]: Could you extract code common with HunspellImpl::checkword to a helper?

**Comment by nikita-moor** [comment]
I'll try to do that, but I'm not an expert in C and C++. My patch was no more than simple copy. However it can be an interesting task, not very quick though.

**Comment by nikita-moor** [comment]
* [x] automated tests
* [x] extract code to a helper

**Comment by nikita-moor** [comment]
Dear @phajdan, could you help me with the Windows build? I have no idea why many tests are failing. As I can see from other merge requests, this problem is not specific to my commit.

**Comment by laszlonemeth** [comment]
Nikita, Vitrum: thanks for your contribution!

**Pull request merged**